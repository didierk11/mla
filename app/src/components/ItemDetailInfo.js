import React from "react";
import { formatPrice } from "../utils/utils";


const ItemDetailInfo = (props) => {
  const condition = props.condition;
  const sold_quantity = props.sold_quantity;
  const title = props.title;
  const amount = props.amount;
  const currency = props.currency;
  const decimals = props.decimals;

  return (
    <div className="item-details-info">
      <div className="item-details-new-sold-quantity">
        {condition === "new" ? "Nuevo" : "Usado"} -{" "}
        {sold_quantity} Vendidos
      </div>
      <div className="item-detail-title">{title}</div>
      <div className="item-detail-price">
        $ {formatPrice(amount, currency, "decimal")}
        {decimals > 0 && (
          <sup className="item-details-price-decimal">
            .{decimals}
          </sup>
        )}
      </div>
      <div className="item-detail-buy">
        <button className="item-detail-buy-btn">Comprar</button>
      </div>
    </div>
  );
};

export default ItemDetailInfo;
