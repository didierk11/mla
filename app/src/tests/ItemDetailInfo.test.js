import React from "react";
import { render } from "react-dom";
import ItemDetailInfo from "../components/ItemDetailInfo";
import renderer from "react-test-renderer";
import data from "./dataSets/items.json";

//testing data
// const condition = data.condition;
// const sold_quantity = data.sold_quantity;
// const title = data.title;
// const amount = data.price.amount;
// const currency = data.price.currency;
// const decimals = data.price.decimals;

const {condition, sold_quantity, title} = data;
const {amount, currency, decimals} = data.price;

describe("Snapshot test:", () => {
  it("ItemDetailInfo render correctly", () => {
    const div = renderer
      .create(
        <ItemDetailInfo
          condition={condition}
          sold_quantity={sold_quantity}
          title={title}
          amount={amount}
          currency={currency}
          decimals={decimals}
        />
      )
      .toJSON();
    expect(div).toMatchSnapshot();
  });
});

describe("Smoke test:", () => {
  it("ItemDetailInfo render without crashing", () => {
    const div = document.createElement("div");
    render(
      <ItemDetailInfo
        condition={condition}
        sold_quantity={sold_quantity}
        title={title}
        amount={amount}
        currency={currency}
        decimals={decimals}
      />,
      div
    );
  });
});
