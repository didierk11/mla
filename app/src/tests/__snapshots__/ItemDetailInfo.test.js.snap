// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test: ItemDetailInfo render correctly 1`] = `
<div
  className="item-details-info"
>
  <div
    className="item-details-new-sold-quantity"
  >
    Nuevo
     -
     
    2000
     Vendidos
  </div>
  <div
    className="item-detail-title"
  >
    Pelota De Futbol Nº 5 + Calidad 350 Gramos Cámara De Látex
  </div>
  <div
    className="item-detail-price"
  >
    $ 
    949
    <sup
      className="item-details-price-decimal"
    >
      .
      9
    </sup>
  </div>
  <div
    className="item-detail-buy"
  >
    <button
      className="item-detail-buy-btn"
    >
      Comprar
    </button>
  </div>
</div>
`;
